File: Submissions/540_Single_Element_in_a_Sorted_Array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int l = 0, r = nums.size()-1, mid = (l+r)/2;
        while(l<r) {
            mid = (l+r)/2;
            if(nums[mid] == nums[mid+1]) {
                if(mid%2==0) l = mid;
                else r = mid-1;
            } else if(nums[mid] == nums[mid-1]) {
                if(mid%2==0) r = mid;
                else l = mid+1;
            } else {
                l = r = mid;
                break;
            }

        }
        return nums[l];
    }
};

File: Kotlin/540_Single_Element_in_a_Sorted_Array.kt
class Solution {
    fun singleNonDuplicate(nums: IntArray): Int {
        var lo = 0
        var hi = nums.size - 1
    
        while (lo < hi) {
            var mid = lo + (hi - lo) / 2
            if (mid % 2 == 1) --mid

            if (nums[mid] != nums[mid + 1]) hi = mid
            else lo = mid + 2
        }

        return nums[lo]
    }
}